### PR TITLE
Snowfakery Version 2.4

### PR DESCRIPTION
**Pull request by prescod**
@pconrad-sfdo In this case the changes are almost entirely docs so you can be the formal reviewer as well please.

### CODE REVIEW

**Comment by davisagli** [comment]
@prescod are you intending to merge #173 before or after this?

**Comment by prescod** [comment]
I turned off auto-merge so that I can merge #574 first.

File: HISTORY.md [hunk 1]
```diff
@@ -4,6 +4,14 @@ In the beginning, programmers created the databases. Now the databases were form
 
 And so [Salesforce.org](http://salesforce.org/) said “Let there be data,” and there was Snowfakery. And it was good.
 
+## Snowfakery 2.4
+
+The Salesforce load order now takes into account the order in which
+object templates were declared in a recipe. The other considerations
```
Comment (pconrad-sfdo) [inline HISTORY.md:10]:
```suggestion
object template declarations in the recipe. The other considerations
```

File: HISTORY.md [hunk 2]
```diff
@@ -4,6 +4,14 @@ In the beginning, programmers created the databases. Now the databases were form
 
 And so [Salesforce.org](http://salesforce.org/) said “Let there be data,” and there was Snowfakery. And it was good.
 
+## Snowfakery 2.4
+
+The Salesforce load order now takes into account the order in which
```
Comment (pconrad-sfdo) [inline HISTORY.md:9]:
```suggestion
The Salesforce load order now follows the order of
```

**Review by pconrad-sfdo — APPROVED** [review]

File: HISTORY.md (hunk 1 above)
Comment (pconrad-sfdo) [reply HISTORY.md:10]: I believe there's one recipe when you're running Snowfakery and that it now determines the load order. If I'm wrong, my suggestions are likely wrong too. Let me know.

**Review by pconrad-sfdo — DISMISSED** [review]

**Review by pconrad-sfdo — APPROVED** [review]

**Pull request merged**